src/lib/core/ipc.js

```diff
@@ -38,6 +38,10 @@ class IPC {
     ipc.connectTo('embark', this.socketPath, connecting);
   }
 
+  rawApi() {
```

Comment:
This might read better as a `get` instead.

Reply:
Could you explain a bit as to why we need to access the underlying API to begin with? Are the current `.request()` and `.on()` methods not sufficient? Maybe instead we should extend those out instead of exposing the underlying API. That way, if we need to swap out `node-ipc`, it will be less coupled.

Reply:
See https://github.com/embark-framework/embark/pull/1213#discussion_r243150266. `.request()` wraps message with `require('flatted/cjs').stringify` and `.on()` enforces assumptions about the structure of the message.

It would be feasible to implement methods like `rawRequest` and `rawOn`, or something like that, but it seemed simpler to expose the "raw" `ipc` object.

A getter method for `ipc.api` might be nicer than `ipc.rawApi()`; I don't have a strong preference either way, but definitely worthy of consideration.

Reply:
Since babel is in place, we could also skip the getter and use an ["instance property"](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties).

Reply:
I was thinking maybe could pass in an overload of `IPC.on()`, maybe it could accept a `raw` parameter that passes back an untransformed message.

Reply:
The instance properly looks like it could definitely work, although again, I think we should prefer to not expose it at all due to coupling concerns.

Reply:
I made some effort to reduce the coupling concerns, both generally and specifically for this use case. See the WIP commit I just pushed: https://github.com/embark-framework/embark/pull/1213/commits/b9d40d9f66b2174da62626783175c39f08b8d490.

Reply:
The commit mentioned in the previous comment is a bit out of date at this point, probably better to just look at what all is squashed into the one commit for this PR.